### ISSUE TITLE
Crash when plugin tries to run code

### ISSUE THREAD

**Issue by anitagraser** [Teakinboyewa/SpatialAnalysisAgent#19]
I downloaded church POIs using your other plugin. Now, I'm testing the analysis agent. It manages to create a geoprocessing workflow but then it crashes QGIS: 

![image](https://github.com/user-attachments/assets/0d095f26-4ae0-4bdf-b0f0-fe03caaa36df)

**Comment by yumemio** [comment]
Throwing in my two cents... :coin: :coin: 

From the screenshot, I can see that the plugin was running code involving `gdal:buffervectors` when the crash occurred. Can’t say for sure without seeing the auto-generated code (actually it might be difficult to grab that code before QGIS crashes), perhaps it's related to [how the Python binding of GDAL may not handle exceptions well in certain scenarios](https://gdal.org/en/latest/api/python/python_gotchas.html#python-crashes-or-throws-an-exception-if-you-use-an-object-after-deleting-a-related-object)?

From what I’ve read, the Python GDAL bindings library can sometimes crash (instead of raising an error) if used incorrectly. This plugin is designed to catch errors and refine the code when GPT-4o makes a mistake, but it won't be able to gracefully handle cases where GDAL causes the Python runtime itself to crash.

**Comment by anitagraser** [comment]
Yes, catching errors before they crash QGIS is definitely a challenge.

**Comment by Teakinboyewa** [comment]
Thank you @anitagraser.
Crashing errors mostly occurs when the plugin uses gpkg files. We are working hard to address this issue.

**Comment by GIBDUSC** [comment]
Yes, for now please use shapefile instead of gpkg files to avoid the crashing.
